### PR TITLE
Signup: add `designFirstSignup` AB test

### PR DESCRIPTION
**Pull request by michaeldcain**
This adds the test stub for the themes-first Signup test. The test is set to the future with a 0/100 split. The test will split traffic from the default `onboarding` Signup once enabled.

Epic: Automattic/zelda-private#119
Requires: Automattic/wp-calypso#36825

**To Test:**
- visit `/start/`
- use the AB tool to select the `designFirst` variant of the `designFirstSignup` test
- on reload, you should see the theme step of the `design-first` Signup flow

### CODE REVIEW

File: client/lib/abtest/active-tests.js [hunk 1]
```diff
@@ -116,6 +106,15 @@ export default {
...
 		localeTargets: 'any',
 	},
+	designFirstSignup: {
+		datestamp: '20201029',
+		variations: {
+			designFirst: 0,
+			default: 100,
+		},
+		defaultVariation: 'default',
+		allowExistingUsers: true,
```
Comment (adamziel) [inline client/lib/abtest/active-tests.js]: Is `allowExistingUsers: true` intentional here?

File: client/signup/config/flows.js [hunk 2]
```diff
@@ -88,6 +89,13 @@ function filterDestination( destination, dependencies ) {
 }
 
 function getDefaultFlowName() {
+	if (
+		config.isEnabled( 'signup/onboarding-flow' ) &&
+		abtest( 'designFirstSignup' ) === 'designFirst'
```
Comment (adamziel) [inline client/signup/config/flows.js]: Is this the right place to do the test split? My concern is getting skewed assignment. Do we have any direct links to `/start/onboarding`?

File: client/signup/step-wrapper/index.jsx [hunk 3]
```diff
@@ -71,7 +71,7 @@ class StepWrapper extends Component {
 						flowName={ this.props.flowName }
 						stepName={ this.props.stepName }
 						labelText={ this.props.skipLabelText }
-						cssClass={ !! this.props.skipHeadingText && ' navigation-link--has-skip-heading ' }
+						cssClass={ ! this.props.skipHeadingText && ' navigation-link--has-skip-heading ' }
```
Comment (adamziel) [inline client/signup/step-wrapper/index.jsx]: this seems unrelated to the rest of the diff - is it intentional or a debugging artifact?

**Comment by adamziel** [comment]
Btw the updated css is quite pretty!

**Comment by adamziel** [comment]
How does this relate to #36825? Should we close one of these?

**Comment by michaeldcain** [comment]
> How does this relate to #36825? Should we close one of these?

I merged that branch into this one for testing. I'll address the feedback in #36825 first and then rebase here.

**Comment by adamziel** [comment]
moving to blocked for now

**Comment by michaeldcain** [comment]
We've decided to shelve this for at least a few sprints, until 0-FSE milestone 2 or 3.
